### PR TITLE
put block/unblock network function into util

### PR DESCRIPTION
**Pull request by freehan**

### CODE REVIEW

**Comment by freehan** [comment]
@bprashanth Not sure if this is what you wanted. Or we can 

```go
type netsplitHelper struct{from string, to string}
func (n netsplitHelper) block()
func (n netsplitHelper) unblock()
```

**Comment by bprashanth** [comment]
Great thanks! I'll take a look later today

File: test/e2e/util.go [hunk 1]
```diff
@@ -2789,3 +2789,51 @@ func getPodLogsInternal(c *client.Client, namespace, podName, containerName stri
...
+// Suggested usage pattern:
+//
+// func foo() {
+//	...
+//	defer unblockNetwork(from, to)
+//	blockNetwork(from, to)
+//	...
+// }
+//
+func blockNetwork(from string, to string) {
```
Comment (bprashanth) [inline test/e2e/util.go]: add a word of caution around using dns names?

**Comment by bprashanth** [comment]
I had initially suggested a type so you can do something like https://github.com/kubernetes/kubernetes/blob/master/test/e2e/daemon_restart.go#L78, but eitherway is fine. If you want to leave it as is, this LGTM

File: test/e2e/util.go [hunk 2]
```diff
@@ -2789,3 +2789,51 @@ func getPodLogsInternal(c *client.Client, namespace, podName, containerName stri
...
+	Logf("block network traffic from %s to %s", from, to)
+	iptablesRule := fmt.Sprintf("OUTPUT --destination %s --jump REJECT", to)
+	dropCmd := fmt.Sprintf("sudo iptables --insert %s", iptablesRule)
+	if result, err := SSH(dropCmd, from, testContext.Provider); result.Code != 0 || err != nil {
```
Comment (bprashanth) [inline test/e2e/util.go]: nit: probably 100% of the people who use this are going to poll till node NotReady right after, and then poll till NodeReady after recovery, so maybe encapsulate that too?

**Comment by freehan** [comment]
@k8s-bot test this issue: #21261

**Pull request merged**